manim/utils/simple_functions.py

```diff
@@ -11,7 +11,7 @@
 
 
 from functools import lru_cache
-from typing import Callable
+from typing import Callable, overload
 
 import numpy as np
 from scipy import special
```

Comment:
I can't directly add a suggestion block for `binary_search`, but, instead of using `int | float`, we prefer using just `float`. Could you please modify all the parameter and return types to use simply `float`?